Generative-AI/Cold-Email-Generator/app/app.py

```diff
@@ -195,66 +186,75 @@ def create_streamlit_app(llm, portfolio, clean_text):
     st.markdown("<div class='subtitle'>Effortlessly craft professional cold emails for job applications based on job postings.</div>", unsafe_allow_html=True)
 
-
     st.markdown("<div class='container'>", unsafe_allow_html=True)
 
-
     user_name = st.text_input("Enter your name:", value=" ")
     user_about = st.text_area(
         "Enter a brief description about yourself:",
         value=" "
     )
 
-
     url_input = st.text_input("Enter a Job Post URL:", value=" ")
 
-
     st.subheader("Enter Your Skills and Portfolio Links")
     skills_input = st.text_area("Enter your skills (comma separated):", value="")
     links_input = st.text_area("Enter your portfolio links (comma separated):", value="")
 
-
     submit_button = st.button("Submit", key='submit_button', help="Click to generate the cold email")
 
-
     if submit_button:
         try:
             skills_list = [skill.strip() for skill in skills_input.split(",")]
             links_list = [link.strip() for link in links_input.split(",")]
 
-
             portfolio.add_to_portfolio(skills_list, links_list)
 
-
             loader = WebBaseLoader([url_input])
             data = clean_text(loader.load().pop().page_content)
             jobs = llm.extract_jobs(data)
 
+            # Initialize an empty email variable for editing
+            email_content = ""
 
             for job in jobs:
                 job_skills = job.get('skills', [])
                 links = portfolio.query_links(job_skills)
-                email = llm.write_mail(job, links, user_name, user_about)
-                st.markdown(f"<div class='code-block'><pre>{email}</pre></div>", unsafe_allow_html=True)
+                email_content = llm.write_mail(job, links, user_name, user_about)
+                st.markdown(f"<div class='code-block'><pre>{email_content}</pre></div>", unsafe_allow_html=True)
 
+            # Add a text area for manual editing of the generated email
+            edited_email = st.text_area("Edit the generated email:", value=email_content, height=200)
 
-        except Exception as e:
-            st.error(f"An Error Occurred: {e}")
+            # Options to save as draft, download, or copy to clipboard
+            col1, col2, col3 = st.columns(3)
 
+            with col1:
+                if st.button("Save as Draft"):
+                    st.session_state['draft_email'] = edited_email
+                    st.success("Email saved as draft!")
 
-    st.markdown("</div>", unsafe_allow_html=True)
+            with col2:
+                if st.button("Download Email"):
+                    # Save the email as a text file for download
+                    st.download_button("Download Email", edited_email, file_name="email.txt", mime="text/plain")
 
+            with col3:
+                if st.button("Copy to Clipboard"):
+                    st.markdown(f"<script>navigator.clipboard.writeText(`{edited_email}`);</script>")
+                    st.success("Email copied to clipboard!")
```

Comment:
Try adding a few seconds of delay before the message disappears in the event of "Copy to Clipboard". Otherwise looks good. Update this and I will approve the PR. Thank you.